src/Microsoft.AspNetCore.Mvc.TagHelpers/SelectTagHelper.cs

```diff
@@ -70,6 +72,12 @@ public override void Init(TagHelperContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            if (For == null && Items != null)
+            {
+                context.Items[typeof(SelectTagHelper)] = null;
```

Comment:
Add a comment indicating this informs contained elements that they're running w/in a targeted `<select>` element. Otherwise someone will remove this.